nautobot/utilities/templatetags/buttons.py

```diff
@@ -29,7 +29,10 @@ def _get_viewname(instance, action):
 
 @register.inclusion_tag("buttons/clone.html")
 def clone_button(instance):
-    url = reverse(_get_viewname(instance, "add"))
+    try:
+        url = reverse(_get_viewname(instance, "add"))
+    except NoReverseMatch:
+        return {"url": None}
```

Comment:
Sorry to be pedantic at the last minute, but I'm seeing code duplication here. Two things:

1.  We already have a helper function/template tag that does this found at `nautobot.utilities.templatetags.helpers.validated_viewname()` which returns `None` if the view name cannot be reversed, so let's please use that here (obviously, it will also need to be imported first):
```suggestion
    url = validated_viewname(instance, "add")
```

2. While we're at it, let's please get rid of the `_get_viewname()` function in this module.

Reply:
I don't think there's any sense in using `validated_viewname` since it just returns the url name ("dcim:device_add"), not the actual URL ("/dcim/devices/add/") and I still need to reverse it to use it. But I can definitely use `nautobot.utilities.utils.get_route_for_model()` at least in place of `_get_viewname`.

Reply:
Yeah, you're right. I should not do code review at 4.30am.